speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Mediator.inl

```diff
@@ -140,6 +142,41 @@ tParticipantControl Mediator<DecisionMaker>::GetPControlSettings()
     return m_decisionMaker.Config.GetPControlSettings();
 }
 
+/// @brief  Sets the decision thresholds to what is in the xml file
+/// @param  p_filePath The path to the xml file
+/// @return The new set threshold values
+template <typename DecisionMaker>
+tThreshold Mediator<DecisionMaker>::SetThresholdSettings(const char* p_filePath)
```

Comment:
Just pass a `tThreshold` struct into this and set the mediator value, no need to read a file here. A setter also should just return void.